ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/bridge/validate/ConflictCheckWithCommittedImpl.scala

```diff
@@ -134,54 +135,88 @@ private[validate] class ConflictCheckWithCommittedImpl(
       completionInfo: CompletionInfo,
   )(implicit
       contextualizedErrorLogger: ContextualizedErrorLogger,
       loggingContext: LoggingContext,
-  ): EitherT[Future, Rejection, Unit] = {
-    // Note that the validation fails fast on the first unknown/invalid contract.
-    disclosedContracts.foldLeft(EitherT(Future.successful[Validation[Unit]](Right(())))) {
-      case (f, provided) =>
-        f.flatMapF { _ =>
-          indexService
-            .lookupContractStateWithoutDivulgence(provided.unversioned.contractId)
-            .map {
-              case ContractState.Archived | ContractState.NotFound =>
-                // Disclosed contract was archived or never existed
-                Left(UnknownContracts(Set(provided.unversioned.contractId))(completionInfo))
-              case ContractState.Active(contractInstance, ledgerEffectiveTime) =>
-                sameContractData(contractInstance, ledgerEffectiveTime, provided).left.map {
-                  errMessage =>
-                    logger.info(errMessage)
-                    DisclosedContractInvalid(provided.unversioned.contractId, completionInfo)
-                }
+  ): EitherT[Future, Rejection, Unit] =
+    // Validation fails fast on the first unknown/invalid contract.
+    disclosedContracts.toList.collectFirstSomeM { disclosedContract =>
+      val validation = for {
+        _ <- validateDisclosedContractPayload(disclosedContract, completionInfo)
+        _ <- validateDriverMetadataContractId(completionInfo, disclosedContract)
+      } yield ()
+
+      validation.value.map(_.left.toOption)
+    }.invertToEitherT
+
+  private def validateDisclosedContractPayload(
+      provided: Versioned[DisclosedContract],
+      completionInfo: CompletionInfo,
+  )(implicit
+      contextualizedErrorLogger: ContextualizedErrorLogger,
+      loggingContext: LoggingContext,
+  ): EitherT[Future, Rejection, Unit] =
+    EitherT(
+      indexService
+        .lookupContractStateWithoutDivulgence(provided.unversioned.contractId)
+        .map {
+          case ContractState.Archived | ContractState.NotFound =>
+            // Disclosed contract was archived or never existed
+            Left(UnknownContracts(Set(provided.unversioned.contractId))(completionInfo))
+          case ContractState.Active(contractInstance, ledgerEffectiveTime) =>
+            sameContractData(contractInstance, ledgerEffectiveTime, provided).left.map {
+              errMessage =>
+                logger.info(errMessage)
+                DisclosedContractInvalid(provided.unversioned.contractId, completionInfo)
             }
         }
-    }
-  }
+    )
+
+  private def validateDriverMetadataContractId(
+      completionInfo: CompletionInfo,
+      provided: Versioned[DisclosedContract],
+  )(implicit
+      contextualizedErrorLogger: ContextualizedErrorLogger
+  ): EitherT[Future, Rejection, Unit] =
+    EitherT
+      .fromEither[Future](
+        ContractId.V1
```

Comment:
So 'driverMetadata' is essentially some contractId?

Reply:
Only for the scope of Sandbox-on-X. This is not a technical requirement nor does it serve any purpose outside testing the conformance tests (see the ones introduced here)

Reply:
Do you see any issue with this?